pilot/pkg/model/push_context.go

```diff
@@ -182,7 +182,14 @@ type XDSUpdater interface {
 	// For each cluster and hostname, the full list of active endpoints (including empty list)
 	// must be sent. The shard name is used as a key - current implementation is using the
 	// registry name.
-	EDSUpdate(shard, hostname string, namespace string, entry []*IstioEndpoint) error
+	EDSUpdate(shard, hostname string, namespace string, entry []*IstioEndpoint)
+
+	// EDSUpdate is called when the list of endpoints or labels in a Service is changed.
+	// For each cluster and hostname, the full list of active endpoints (including empty list)
+	// must be sent. The shard name is used as a key - current implementation is using the
+	// registry name.
+	// Note: the difference with `EDSUpdate` is that it only update the cache rather than requesting a push
+	EDSUpdateCacheOnly(shard, hostname string, namespace string, entry []*IstioEndpoint)
```

Comment:
nit: Should we call `EDSCacheUpdate` so that it is super clear?

Reply:
or just UpdateEndpointCache / UpdateEdsCache?

Reply:
looks better

Reply:
renamed to `EDSCacheUpdate` to keep naming consistent with other methods